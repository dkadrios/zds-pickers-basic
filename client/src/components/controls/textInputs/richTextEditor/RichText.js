import RichText from 'react-quill'
import { formats, modules } from './reactQuillToolbar'

export default props => (
  <RichText
    formats={formats}
    modules={modules}
    theme="snow"
    {...props}
  />
)
