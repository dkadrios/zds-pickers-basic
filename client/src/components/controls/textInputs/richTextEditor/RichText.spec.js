import RichText from './RichText'

describe('<RichText />', () => {
  beforeEach(jest.clearAllMocks)
  const value = '<h1>header</h1><p>paragraph <strong>strong</strong> text</p>'
  const onChange = jest.fn()
  const props = { value, onChange }

  it('should render', () => {
    shallowExpect(RichText)(props).toMatchSnapshot()
  })
})
