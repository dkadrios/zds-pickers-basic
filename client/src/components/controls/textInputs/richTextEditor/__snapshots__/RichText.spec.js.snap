// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichText /> should render 1`] = `
<Quill
  formats={
    Array [
      "header",
      "bold",
      "italic",
      "underline",
      "strike",
      "list",
      "bullet",
      "indent",
      "link",
      "video",
    ]
  }
  modules={
    Object {
      "toolbar": Array [
        Array [
          Object {
            "header": Array [
              false,
              1,
              2,
              3,
              4,
              5,
              6,
            ],
          },
        ],
        Array [
          "bold",
          "italic",
          "underline",
          "strike",
        ],
        Array [
          Object {
            "list": "ordered",
          },
          Object {
            "list": "bullet",
          },
          Object {
            "indent": "-1",
          },
          Object {
            "indent": "+1",
          },
        ],
        Array [
          "link",
          "video",
        ],
      ],
    }
  }
  onChange={[MockFunction]}
  theme="snow"
  value="<h1>header</h1><p>paragraph <strong>strong</strong> text</p>"
/>
`;
